BrainPortal/app/models/tool_config.rb

```diff
@@ -326,22 +326,45 @@ def cbrain_task_class
 
   # Returns an array of full paths to the Singularity overlay files that
   # need to be mounted, as configured by the admin. Some of them might
-  # be patterns and will need to be resolved at run time.
+  # be patterns and will need to be resolved at run time. The dsl is
+  #      # A file located on the cluster
+  #        file:/path/to/file.squashfs
+  #      # All overlays configured for DP 1234
+  #         dp:1234
+  #      # My special overlay in CBRAIN
+  #         userfile:1234
   def singularity_overlays_full_paths
     specs = self.singularity_overlays_specs.presence
     return [] if specs.blank?
 
-    specs = specs.split(/[\s,]+/).map(&:presence).compact
+    # break onto commands based on comas, whitespace or comments
+    specs = specs.split(/([\s,]|#.*$)+/).map(&:presence).compact
```

Comment:
You're breaking text on hash signs here, that's not right. If you want to support embedded comments, you must split by lines first. This code will crash on a string like "hello#bye".

Reply:
The new spec should not allow several specifications on the same line, only one per line.

Reply:
ok got. I thought 


```# folder a files
file:/a/1.sqs file:/a/2.sqs file:/a/3.sqs
# folder b files
/b/0.sqs
# data providers
dp:3  dp:4```

could be of use, but can limit to one liners

Reply:
1. basically I just try conform to the ticket's requirement for supporting old legacy specification format, which I believed allowed for multi spec lines. Alternatively, I can try to determine is spec of old or new format, e.g. by presence of colon (assuming no file name includes colon) and then apply either or validation function or new one. Not sure would it be overcomplication.
2. Advantage of statement per line notation, spaces in file names can be allowed `  /a/b/my beatiful brain.sqs` 
3. Would you like to have trailing comments `file:/x/x.sqs # that's file with my brain scans`
@prioux

Reply:
Ok more clearly:

1. one spec per line only
2. comments allowed on standalone lines and on spec lines too
3. don't allow spaces in the file specs

```
# comment
file:/a/b/c.sqs
file:/a/b/d.sqs #with comment
dp:3
dp:MainStore # by name shoudl work already
userfile:12455  # no support for 'by name' here though, must be an ID
```

Reply:
I suggest making an intermediate-level method `parsed_overlay_specs` that takes the text and returns something like

```
overlay_specs = tc.parsed_overlay_specs
-> [ [ 'file', '/a/b/c.sqs'], [ 'dp', '3' ] ... ]
```

Reply:
ok